spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CqlSessionFactoryBean.java

```diff
@@ -138,6 +152,19 @@ public void setContactPoints(String contactPoints) {
 		this.contactPoints = contactPoints;
 	}
 
+	/**
+	 * Set a comma-delimited string of the contact points (hosts) with port to connect to. Default is null, meaning that
+	 * this method will be ignored and standard {@link CqlSessionFactoryBean#setContactPoints(String)} and
+	 * {@link CqlSessionFactoryBean#setPort(int)} will apply. If set, it will have priority over the aforementioned
+	 * methods.
+	 *
+	 * @param contactPointsWithPort the contact points with port used by the new cluster. The format of this setting
+	 *          should be following: host:port,host2:port2
+	 */
+	public void setContactPointsWithPort(String contactPointsWithPort) {
```

Comment:
I've added a new method that works in a backward-compatible way:
- if the old solution: `addContactPoints` and `addPort` is used it works the same way
- if someone intends to specify the port per contact point, the new API can be used and it has priority over the old API.
Also, I am not sure about the type of the argument. Maybe it will be better to accept here a list of addresses? 
Or even list of tuples (or something similar) with `String host` and `int port`?
Right now I've created one string argument to have this API unified with the existing `setContactPoints` but because of the fact that `String` is used, there is validation required (see my 2nd comment).
wdyt?